chia/wallet/trading/offer.py

```diff
@@ -114,22 +121,18 @@ def get_offered_coins(self) -> Dict[Optional[bytes32], List[Coin]]:
             )[0].puzzle_reveal.to_program()
 
             # Determine it's TAIL (or lack of)
-            matched, curried_args = match_cat_puzzle(parent_puzzle)
-            tail_hash: Optional[bytes32] = None
-            if matched:
-                _, tail_hash_program, _ = curried_args
-                tail_hash = bytes32(tail_hash_program.as_python())
-                offer_ph: bytes32 = construct_cat_puzzle(CAT_MOD, tail_hash, OFFER_MOD).get_tree_hash()
+            asset_id = asset_id_of_puzzle(parent_puzzle)
+            if asset_id is not None:
+                offer_ph: bytes32 = construct_puzzle(self.type_dict[asset_id], asset_id, OFFER_MOD).get_tree_hash()
             else:
-                tail_hash = None
                 offer_ph = OFFER_MOD.get_tree_hash()
 
             # Check if the puzzle_hash matches the hypothetical `settlement_payments` puzzle hash
             if addition.puzzle_hash == offer_ph:
-                if tail_hash in offered_coins:
-                    offered_coins[tail_hash].append(addition)
+                if asset_id in offered_coins:
```

Comment:
From the code right above this, it looks like `asset_id` could be None. Is it safe to enter this block of code if `asset_id` is None (doesn't look like it...)

Should the `if addition.puzzle_hash == offer_ph` check above also check that `asset_id is not None`?

Reply:
No, offered coins is `Dict[Optional[bytes32], List[Coin]`.  `asset_id == None` is representing standard XCH